test/tasks/before-karpenter-suite.yaml

```
@@ -0,0 +1,85 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
+metadata:
+  name: before-karpenter-suite
+spec:
+  description: |
+    Initialize an environment for testing karpenter
+  params:
+    - name: cluster-name
+      default: $(context.taskRun.name)
+    - name: karpenter-version
+      default: latest
+    - name: account-id
+      default: "767520670908"
+  steps:
+  - name: cluster
+    image: docker.io/weaveworks/eksctl:v0.86.0
+    script: |
+      eksctl create cluster -f - << EOF
+      ---
+      apiVersion: eksctl.io/v1alpha5
+      kind: ClusterConfig
+      metadata:
+        name: $(params.cluster-name)
+        version: 1.22
+        tags:
+          karpenter.sh/discovery: $(params.cluster-name)
+      managedNodeGroups:
+        - instanceType: m5.large
+          amiFamily: AmazonLinux2
+          name: $(params.cluster-name)-system-pool
+          desiredCapacity: 1
+          minSize: 1
+          maxSize: 1
+      iam:
+        withOIDC: true
+      EOF
+  - name: kubeconfig
+    image: docker.io/njtran/scratch:latest
+    script: |
+        aws eks update-kubeconfig --name $(params.cluster-name)
+
+        TEMPOUT=$(mktemp)
+        curl -fsSL https://karpenter.sh/"$(params.karpenter-version)"/getting-started/getting-started-with-eksctl/cloudformation.yaml  > $TEMPOUT \
+        && aws cloudformation deploy \
+          --stack-name "Karpenter-$(params.cluster-name)" \
```

This is going to collide with our karpenter installation if using the same cluster name as the management cluster. Can we make it the following? It may be better to change up the host-cluster setup too so that we don't run into the chance of IAM Role name conflicts.
```suggestion
          --stack-name "KarpenterTesting-$(params.cluster-name)" \
```